test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs

```diff
@@ -0,0 +1,235 @@
+using Dynamo.Graph.Workspaces;
+using Dynamo.Wpf.Extensions;
+
+namespace DynamoCoreWpfTests
+{
+    class PythonMigrationViewExtensionTests : DynamoTestUIBase
+    {
+        private PythonMigrationViewExtension viewExtension = new PythonMigrationViewExtension();
+
+        private string PackagesDirectory { get { return Path.Combine(GetTestDirectory(this.ExecutingDirectory), "pkgs"); } }
+
+        protected override DynamoModel.IStartConfiguration CreateStartConfiguration(IPathResolver pathResolver)
+        {
+            return new DynamoModel.DefaultStartConfiguration()
+            {
+                PathResolver = pathResolver,
+                StartInTestMode = true,
+                GeometryFactoryPath = this.preloader.GeometryFactoryPath,
+                ProcessMode = TaskProcessMode.Synchronous,
+                Preferences = new PreferenceSettings() { CustomPackageFolders = new List<string>() { this.PackagesDirectory } }
+            };
+        }
+
+        /// <summary>
+        /// This test is created to check if the extension displays a dialog to the user
+        /// when opening a saved dyn file that contains python nodes with their engine set to `IronPython2`
+        /// </summary>
+        [Test]
+        public void WillDisplayDialogWhenOpeningGraphWithIronPythonNodes()
+        {
+            // Arrange
+            RaiseLoadedEvent(this.View);
+            var extensionManager = View.viewExtensionManager;
+            extensionManager.Add(viewExtension);
+
+            // Act
+            // open file
+            var examplePath = Path.Combine(UnitTestBase.TestDirectory, @"core\python", "python.dyn");
+            Open(examplePath);
+
+            var isIronPythonDialogOpen = this.View.OwnedWindows.Cast<Window>().Any(x=>x.GetType() == typeof(IronPythonInfoDialog));
+        
+            // Assert
+            Assert.IsTrue(isIronPythonDialogOpen);
+        }
+
+        /// <summary>
+        /// This test is created to check if a notification is logged to the Dynamo Logger 
+        /// when adding a using the IronPython engine
+        /// </summary>
+        [Test]
+        public void WillLogNotificationWhenAddingAnIronPythonNode()
+        {
+            // Arrange
+            string pythonNodeName = "Python Script";
+            RaiseLoadedEvent(this.View);
+            var raisedEvents = new List<string>();
+
+            // Act
+            // open file
+            this.ViewModel.Model.Logger.NotificationLogged += delegate (Dynamo.Logging.NotificationMessage obj)
+            {
+                raisedEvents.Add(obj.Sender);
+            };
+
+            var nodesCount = this.ViewModel.CurrentSpace.Nodes.Count();
+
+            this.ViewModel.ExecuteCommand(new DynamoModel.
+                CreateNodeCommand(Guid.NewGuid().ToString(), pythonNodeName, 0, 0, false, false));
+
+            // Assert
+            Assert.AreNotEqual(this.ViewModel.CurrentSpace.Nodes.Count(), nodesCount);
```

Comment:
move `this.ViewModel.CurrentSpace.Nodes.Count()` to act bit above and save to a variable and assert that it's `old count + 1`